dotnet/src/SemanticKernel.Abstractions/Contents/AnnotationContent.cs

```diff
@@ -12,25 +13,60 @@ namespace Microsoft.SemanticKernel.Agents;
 public class AnnotationContent : KernelContent
 {
     /// <summary>
-    /// The file identifier.
+    /// The referenced file identifier.
+    /// </summary>
+    [JsonIgnore]
+    [Obsolete("Use `ReferenceId` property instead.  This method will be removed after June 1st 2025.")]
+    public string? FileId
+    {
+        get => this.ReferenceId;
+        init => this.ReferenceId = value ?? string.Empty;
+    }
+
+    /// <summary>
+    /// The citation label in the associated response.
+    /// </summary>
+    [JsonIgnore]
+    [Obsolete("Use `Label` property instead.  This method will be removed after June 1st 2025.")]
+    public string Quote => this.Label;
+
+    /// <summary>
+    /// Identifies the referenced resource according to its <see cref="Kind"/>.
     /// </summary>
     [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
-    public string? FileId { get; init; }
+    public string ReferenceId { get; init; } = string.Empty;
 
     /// <summary>
-    /// The citation.
+    /// The citation label in the associated response.
     /// </summary>
-    public string Quote { get; init; } = string.Empty;
+    /// <remarks>
+    /// This is the citation referebce that will be displayed in the response.
+    /// </remarks>
+    public string Label { get; init; } = string.Empty;
+
+    /// <summary>
+    /// Describes the annotation kind.
+    /// </summary>
+    /// <remarks>
+    /// Provides context for using <see cref="ReferenceId"/>.
+    /// </remarks>
+    public AnnotationKind Kind { get; init; }
+
+    /// <summary>
+    /// The title of the annotation reference.
+    /// </summary>
+    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+    public string? Title { get; init; }
 
     /// <summary>
     /// Start index of the citation.
     /// </summary>
-    public int StartIndex { get; init; }
+    public int? StartIndex { get; init; }
```

Comment:
Suggest adding UnitTests ensuring the serialization of all this class.

```suggestion
    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
    public int? StartIndex { get; init; }
```

Reply:
Good call on `JsonIgnore`.

> NOTE: Unit tests performing serialization already present